backend/src/Designer/TypedHttpClients/ImageClient/ImageClient.cs

```diff
@@ -18,29 +15,25 @@ public async Task<ImageUrlValidationResult> ValidateUrlAsync(string url)
     {
         try
         {
-            // Send a HEAD request to the URL to check if the resource exists and fetch the headers
             using var request = new HttpRequestMessage(HttpMethod.Head, url);
             var response = await _httpClient.SendAsync(request, cancellationToken: default);
 
-            // If the response status is not successful return NotValidUrl
             if (!response.IsSuccessStatusCode)
             {
-                return ImageUrlValidationResult.NotValidUrl;
+                return ImageUrlValidationResult.NotValidImage;
             }
-            var contentType = response.Content.Headers.ContentType.MediaType;
-            if (!contentType.StartsWith("image/"))
+
+            string contentType = response.Content.Headers.ContentType?.MediaType;
+            if (string.IsNullOrEmpty(contentType) || !contentType.StartsWith("image/"))
             {
-                // If the response did not return an image in its content return NotAnImage
-                return ImageUrlValidationResult.NotAnImage;
+                return ImageUrlValidationResult.NotValidImage;
             }
 
             return ImageUrlValidationResult.Ok;
         }
-        // If the request fails for some other reason return NotValidUrl
-        catch (Exception ex)
-            when (ex is UriFormatException or InvalidOperationException or HttpRequestException)
+        catch
```

Comment:
Are we foreseeing any other errors than `HttpRequestException` here?
Maybe we should let other exceptions propagate upwards for debugging purposes?

```suggestion
        catch (HttpRequestException)
```